### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by ahmedqarmout2**
fixing logger

### CODE REVIEW

File: public/js/admin.js [hunk 1]
```diff
@@ -4,6 +4,25 @@ var autocompleteTopics;
 $(function(){
     /* show the account table by default */
     displayQuestionTable();
+
+        $.ajax({
+            type: 'POST',
+            url: '/login',
+            data: {user:'ad', passwd: 'hi'},
```
Comment (Mohammed-JS-Khan) [inline public/js/admin.js]: What is this whole block? And why are there hardcoded credentials?

**Review by Mohammed-JS-Khan — APPROVED** [review]
Looks good, just one little goof i think. But i may be wrong.

File: public/js/admin.js (hunk 1 above)
Comment (ahmedqarmout2) [reply public/js/admin.js]: Lol I was testing Darshana stuff. I will remove it. Nice catch

**Review by Mohamed-A-Mohamed — APPROVED** [review]
Looks good aside from the thing Mo pointed out :laughing:

**Review by mehtadarshan1 — APPROVED** [review]
awesome

File: server/common.js [hunk 2]
```diff
@@ -141,7 +141,33 @@ var isEmptyObject = function(obj) {
...
+
+// return the current date with format
+var getDateByFormat = function(format) {
+    return date().format(format);
+}
+
 exports.getDate = getDate;
+exports.getDateByFormat = getDateByFormat;
+
+// This is the function.
```
Comment (esteeroramy) [inline server/common.js]: better comment?

File: server/log.js [hunk 3]
```diff
@@ -18,17 +18,40 @@ You should have received a copy of the GNU General Public License
...
+    var txt = common.formatString('[{0}] log: {1}', [common.getDate(), text]);
+    console.log(txt);
+}
+
+// print an error message
+exports.error = function (text) {
+    init();
+
+    var txt = common.formatString('[{0}] error: {1}', [common.getDate(), text]);
+    console.error(txt);
```
Comment (esteeroramy) [inline server/log.js:39]: do we really need that extra var? can we just do:
``` javascript
console.error(common.formatString('[{0}] error: {1}', [common.getDate(), text]));
```
This also applies to the `console.log()` above

File: server/log.js [hunk 4]
```diff
@@ -18,17 +18,40 @@ You should have received a copy of the GNU General Public License
...
+var init = function () {
+    var currentDate = common.getDateByFormat('YYYY-MM-DD');
+    if (!process.env.DEBUG && dateStamp !== currentDate) {
+        dateStamp = currentDate;
+        logger = fs.createWriteStream(__dirname + '/../logs/'+dateStamp+'.log', {'flags':'a'});
+        process.stdout.write = process.stderr.write = logger.write.bind(logger);
+        process.on(
+            'uncaughtException',
+            function(err)
+            {
```
Comment (esteeroramy) [inline server/log.js]: can we use the standard way we di bracketing here:
``` javascript
function(err) {
    //contents
}
```

**Review by esteeroramy — COMMENTED** [review]
very minor changes to look at here, also I am wondering if we should make some sort of translations file in order to standardize the logs and error messages we have, what do you think? good work overall :)

File: server/log.js (hunk 3 above)
Comment (ahmedqarmout2) [reply server/log.js:39]: Not now. But I am planing to add more information to the logs, such as the userId and parts of the me object, which will make this line much longer.

File: server/log.js (hunk 4 above)
Comment (ahmedqarmout2) [reply server/log.js]: yeah my bad. will fix

File: server/common.js (hunk 2 above)
Comment (ahmedqarmout2) [reply server/common.js]: what the hell? did I actually right this?! will fix

**Pull request merged**